src/internal/components/selectable-item/index.tsx

```diff
@@ -3,11 +3,63 @@
 import React, { useLayoutEffect, useRef } from 'react';
 import clsx from 'clsx';
 
+import { getAnalyticsMetadataAttribute } from '@cloudscape-design/component-toolkit/internal/analytics-metadata';
+
 import { BaseComponentProps, getBaseProps } from '../../base-component';
 import { HighlightType } from '../options-list/utils/use-highlight-option.js';
+import { GeneratedAnalyticsMetadataSelectableItemSelect } from './analytics-metadata/interfaces';
 
+import optionAnalyticsSelectors from './../option/analytics-metadata/styles.css.js';
+import analyticsSelectors from './analytics-metadata/styles.css.js';
 import styles from './styles.css.js';
 
+export interface ItemDataAttributes {
```

Comment:
For this part of the code would it make since to move it in its own file (like a utility file or so) ?

Reasoning behind this: As an open-source contributor I will get confused by trying to read and understand the analytics logic and how it flows.

Reply:
Great idea, I have refactored the code